### ISSUE TITLE
[Question] - Extend some GPU op

### ISSUE THREAD

**Issue by eyalhir74** [tensorflow/tensorflow#72422]
### Issue type

Others

### Have you reproduced the bug with TensorFlow Nightly?

Yes

### Source

source

### TensorFlow version

v2.17.0-rc1-2-gad6d8cc177d 2.17.0

### Custom code

Yes

### OS platform and distribution

Ubuntu 20.04.6 LTS

### Python version

3.9.19

### GPU model and memory

A30

### Current behavior?

I'm running tensorflow_model_server with the sample saved_model_half_plus_two_gpu model. One of the layers there is Mul

"Mul" op: "Mul" input: "a/read" input: "_arg_x_0_0/_7" device: "/job:localhost/replica:0/task:0/device:GPU:0" attr { key: "T" value { type: DT_FLOAT } } attr { key: "_XlaHasReferenceVars" value { b: true } }

I would like to add some custom code before/after the code runs on the GPU (I have a docker with TensorFlow serving, it downloads the TensorFlow code and comiles it via bazel). I am able to add a print in the OpKernel::OpKernel code, but I would like to do so in the relevant GPU operator compute's function. That I am unable to find/do.

Tried to print/log to file in either of the following places but it did **NOT** work:
- tensorflow/core/util/gpu_kernel_helper.h               in GpuLaunchKernel function
- tensorflow/core/kernels/cwise_ops_common.h     in BinaryOp::Compute

Any idea what am I missing? Obviously my code adds catch as they compile and the code in OpKernel::OpKernel  does work.

### Standalone code to reproduce the issue

```shell
Not relevant...
```

**Comment by tilakrayal** [comment]
@eyalhir74,
Could you please provide more info/context about the above statement which helps to analyse the issue. Thank you!

**Comment by eyalhir74** [comment]
@tilakrayal 
Yes, I would like to try and change a few things in the way TF uses the GPU. I was able to put some custom [NVTX](https://github.com/jrhemstad/nvtx_wrappers/blob/master/nvtx3.hpp) code inside the BaseGPUDevice::Compute code in tensorflow/core/common_runtime/gpu/gpu_device.cc, and it indeed works.\
I'm still looking at why the code in tensorflow/core/util/gpu_kernel_helper.h in GpuLaunchKernel function
did not catch and I'm unable to change the way a CUDA GPU kernel is being launched. Would that code be the right place
to alter CUDA kernel invocations or there is somewhere else?

thanks
Eyal

**Comment by Venkat6871** [comment]
Hi,

Thank you for opening this issue. Since this issue has been open for a long time, the code/debug information for this issue may not be relevant with the current state of the code base.

The Tensorflow team is constantly improving the framework by fixing bugs and adding new features. We suggest you try the latest TensorFlow [version](https://pypi.org/project/tensorflow/) with the latest compatible hardware configuration which could potentially resolve the issue. If you are still facing the issue, please create a new GitHub issue with your latest findings, with all the debugging information which could help us investigate.

Please follow the [release notes](https://github.com/tensorflow/tensorflow/blob/master/RELEASE.md) to stay up to date with the latest developments which are happening in the Tensorflow space.
